ftw/upgrade/command/list_cmd.py

```diff
@@ -0,0 +1,101 @@
+from ftw.upgrade.command.jsonapi import add_json_argument
+from ftw.upgrade.command.jsonapi import add_requestor_authentication_argument
+from ftw.upgrade.command.jsonapi import add_site_path_argument
+from ftw.upgrade.command.jsonapi import error_handling
+from ftw.upgrade.command.jsonapi import with_api_requestor
+from ftw.upgrade.command.terminal import colorize_profile_id
+from ftw.upgrade.command.terminal import colorized_profile_flags
+from ftw.upgrade.command.terminal import colorized_profile_versions
+from ftw.upgrade.command.terminal import print_table
+from ftw.upgrade.command.terminal import TERMINAL
+from ftw.upgrade.command.terminal import upgrade_id_with_flags
+
+
+DOCS = """
+{t.bold}DESCRIPTION:{t.normal}
+    List profiles or upgrades of a Plone site.
+
+{t.bold}LIST PROFILES:{t.normal}
+    Listing profiles lists all profiles which are installed on this Plone \
+site and have at least one upgrade step. The profiles are listed with profile \
+ID, amount of proposed and orphan upgrades for this profile, the title as \
+well as database and filesystem versions.
+
+[quote]
+    $ ./bin/upgrade list --site Plone --profiles
+    $ ./bin/upgrade list --site Plone --profiles --autho admin:admin
+    $ ./bin/upgrade list --site Plone --profiles --autho admin:admin --json
+[/quote]
+
+{t.bold}LIST PROPOSED UPGRADES:{t.normal}
+    Listing proposed upgrades lists all upgrades which are proposed for this \
+Plone site. Only profiles installed on this Plone site are respected.
+
+[quote]
+    $ ./bin/upgrade list --site Plone --proposed
+    $ ./bin/upgrade list --site Plone --proposed --autho admin:admin
+    $ ./bin/upgrade list --site Plone --proposed --autho admin:admin --json
+[/quote]
+""".format(t=TERMINAL).strip()
+
+
+def setup_argparser(commands):
+    command = commands.add_parser(
+        'list',
+        help='List upgrades or profiles.',
+        description=DOCS)
+    command.set_defaults(func=list_command)
+    add_requestor_authentication_argument(command)
+    add_site_path_argument(command)
+    add_json_argument(command)
+
+    group = command.add_mutually_exclusive_group(required=True)
+    group.add_argument('--proposed', '-u',
```

Comment:
I would maybe consider renaming this to `--upgrades`. The basic distinction you also make in the help message above is `List upgrades or profiles` - the fact that it's only *proposed* upgrades and *installed* profiles (with upgrades) is a detail that can be addressed in the help message for those flags, and should also be pretty obvious from the primary use case.

This would in turn allow for nicer short versions of the commands that both match the beginning letter of the long version: `-u` / `--upgrades` and `-p` / `--profiles`